src/engine.c

```diff
@@ -113,51 +113,40 @@ ENGINE_writeFile(ENGINE* engine, char* path, char* buffer, size_t length) {
 
 internal char*
 ENGINE_readFile(ENGINE* engine, char* path, size_t* lengthPtr) {
+  char pathBuf[PATH_MAX];
+
+  if (strncmp(path, "./", 2) == 0) {
+    strcpy(pathBuf, path + 2);
+  } else {
+    strcpy(pathBuf, path);
+  }
+
   if (engine->tar != NULL) {
-    char pathBuf[PATH_MAX];
-    strcpy(pathBuf, "\0");
-    if (strncmp(path, "./", 2) != 0) {
-      strcpy(pathBuf, "./");
-    }
-    strcat(pathBuf, path);
-    mtar_header_t h;
-    int success = mtar_find(engine->tar, pathBuf, &h);
-    if (success == MTAR_ESUCCESS) {
-      ENGINE_printLog(engine, "Reading from bundle: %s\n", path);
-      char* file = readFileFromTar(engine->tar, pathBuf, lengthPtr);
-      if (file == NULL) {
-        ENGINE_printLog(engine, "Error: Couldn't read the data from the bundle.");
-      }
+    ENGINE_printLog(engine, "Reading from bundle: %s\n", pathBuf);
+
+    char* file = NULL;
+    int err = readFileFromTar(engine->tar, pathBuf, lengthPtr, &file);
+    if (err == MTAR_ESUCCESS) {
       return file;
-    } else if (success != MTAR_ENOTFOUND) {
+    } else if (err == MTAR_ENOTFOUND) {
```

Comment:
Turns out, microtar doesn't always use this error code. Running it on the spaceshooter example results in `MTAR_ENULLRECORD` when trying to read config.txt (a file which is explicitly not included in the `game.egg` to test for fallback IO.) I think we are safe to check for both.

Reply:
We might want to get rid of bundle to fs fallback altogether (see 8292dc9)

Reply:
The fallback is intentional, because without it, you can't have local save files or configuration/preference files.

Reply:
Ahh I completely missed that one. What if we have two possible APIs? One to read from assets, so potentially always a RO operation (e.g. bundle, fs, romfs... you name it. We could later use a layer here with something like physfs), and other to read/write from data on the fs?

Reply:
I would quite like to avoid that, for a couple reasons.
1) It complicates the FS api, which is against the design philosophy (and may be considered a breaking change
2) Impractical usage: The programmer likely won't be using a .egg until the end of development, and so once they switch to `game.egg`, they'd have to change their code.

The original design makes the bundle transparent, for easier distribution.

Reply:
I see. Then I guess we could just fallback to fs if tar reading fails for whatever reason, instead of just on `MTAR_ENOTFOUND`?

Reply:
I think that's a suitable solution.